Core/include/Acts/Propagator/Propagator.ipp

```diff
@@ -256,6 +256,10 @@ auto Acts::Propagator<S, N>::makeResult(propagator_state_t state,
   moveStateToResult(state, result);
 
   if (makeCurvilinear) {
+    if (!m_stepper.prepareCurvilinearState(state, m_navigator)) {
+      // information to compute curvilinearState is incomplete.
+      return propagationResult.error();
+    }
```

Comment:
do you think this can be triggered within the stepper? for example during state creation

Reply:
> do you think this can be triggered within the stepper? for example during state creation

The functions called by the eigen stepper implementation could fail in theory e.g. if the extensions are not valid for the "current step", and I think the computation of the magnetic field can potentially fail. In practice it is unlikely.

Reply:
Is it possible to implement this purely inside the stepper without touching the propagator?

to me it looks a bit odd that we have to touch the propagator for this as it looks like a stepper issue to me.

but maybe I am overlooking something

Reply:
Yes, but then curvilinearState would have to become a template, which presumably would mean to add one layer on-top of curvilinearState  e.g.:
```
    template< class T_Navigator, class T_PropagatorState>
    Stepper::curvilinearState(T_PropagatorState prop_state, T_Navigator navigator nav) {
      if (!prepareCurvilinearState(prop_state, nav)) {
        return; // error
      }
      return curvilinearState_impl(prop_state.stepping);
    }
```

Reply:
I see. not sure if we could wire these templates

alternatively we could change the propagator to make a 0 distance step even if we are on target to begin with

any other suggestions @paulgessinger @asalzburger ?